common/flatpak-instance.c

```diff
@@ -313,6 +317,47 @@ flatpak_instance_get_info (FlatpakInstance *self)
   return priv->info;
 }
 
+GStrv
+flatpak_instance_get_run_environ (FlatpakInstance *self)
+{
+  FlatpakInstancePrivate *priv = flatpak_instance_get_instance_private (self);
+
+  return priv->run_environ;
+}
+
+static GStrv
+get_run_environ (const char *dir)
+{
+  g_autofree char *file = NULL;
+  g_autofree char *environ_data = NULL;
+  gsize environ_size;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GPtrArray) env_vars = NULL;
+  int i;
+  gsize env_var_size;
+
+  file = g_build_filename (dir, "run-environ", NULL);
+
+  if (!g_file_get_contents (file, &environ_data, &environ_size, &error))
+    {
+      g_info ("Failed to load run-environ file '%s': %s", file, error->message);
+      return NULL;
+    }
+
+  env_vars = g_ptr_array_new_with_free_func (g_free);
+
+  for (i = 0; i < environ_size - 1; i += env_var_size)
+    {
+      env_var_size = strlen (&environ_data[i]) + 1;
```

Comment:
At first glance you might think this could read-overrun the buffer if the last variable lacks its `\0` terminator. In fact `g_file_get_contents` guarantees to add an extra `\0` at `data[size]`, so that problem can't happen (although that deserves a comment, I think). But if we factor out the parsing into a `flatpak_parse_env_block`, then it will become necessary to be more careful about that, as was done in `flatpak_context_parse_env_block`.

Reply:
This is now resolved through the use of `flatpak_parse_env_block()`.